unix_integration/src/idprovider/interface.rs

```diff
@@ -86,8 +91,52 @@ pub enum AuthCacheAction {
     PasswordHashUpdate { cred: String },
 }
 
+pub struct KeyStore<'a> {
```

Comment:
Can we wrap this in a RwLock, or do something similar? I'm getting errors because I can't pass the keystore across threads. Himmelblau is a bit unique in that we can have multiple backend providers, so I have a main provider that communicates with the child providers. Passing the keystore to the child providers throws the error.

Reply:
What's the error? It might actually need to be Pin to make it work.